pkg/kubelet/eviction/eviction_manager_test.go

```diff
@@ -1784,3 +1810,62 @@ func TestUpdateMemcgThreshold(t *testing.T) {
 	fakeClock.Step(2 * notifierRefreshInterval)
 	manager.synchronize(diskInfoProvider, activePodsFunc)
 }
+
+func TestManagerWithLocalStorageCapacityIsolationOpen(t *testing.T) {
+	podMaker := makePodWithLocalStorageCapacityIsolationOpen
+	summaryStatsMaker := makeDiskStats
+	podsToMake := []podToMake{
+		{name: "empty-dir", requests: newResourceList("", "900Mi", ""), limits: newResourceList("", "1Gi", "")},
+		{name: "container-ephemeral-storage-limit", requests: newResourceList("", "", "900Mi"), limits: newResourceList("", "", "800Mi")},
+		{name: "pod-ephemeral-storage-limit", requests: newResourceList("", "", "1Gi"), limits: newResourceList("", "", "800Mi")},
+	}
+
+	pods := []*v1.Pod{}
+	podStats := map[*v1.Pod]statsapi.PodStats{}
+	for _, podToMake := range podsToMake {
+		pod, podStat := podMaker(podToMake.name, podToMake.priority, podToMake.requests, podToMake.limits, podToMake.memoryWorkingSet)
+		pods = append(pods, pod)
+		podStats[pod] = podStat
+	}
+
+	summaryProvider := &fakeSummaryProvider{result: summaryStatsMaker("1Gi", "200Mi", podStats)}
+
+	config := Config{
+		MaxPodGracePeriodSeconds: 5,
+		PressureTransitionPeriod: time.Minute * 5,
+		Thresholds: []evictionapi.Threshold{
+			{
+				Signal:   evictionapi.SignalAllocatableMemoryAvailable,
+				Operator: evictionapi.OpLessThan,
+				Value: evictionapi.ThresholdValue{
+					Quantity: quantityMustParse("1Gi"),
+				},
+			},
+		},
+	}
+
+	podKiller := &mockPodKiller{}
+	diskGC := &mockDiskGC{err: nil}
+	nodeRef := &v1.ObjectReference{Kind: "Node", Name: "test", UID: types.UID("test"), Namespace: ""}
+	fakeClock := testingclock.NewFakeClock(time.Now())
+	diskInfoProvider := &mockDiskInfoProvider{dedicatedImageFs: false}
+
+	mgr := &managerImpl{
+		clock:                         fakeClock,
+		killPodFunc:                   podKiller.killPodNow,
+		imageGC:                       diskGC,
+		containerGC:                   diskGC,
+		config:                        config,
+		recorder:                      &record.FakeRecorder{},
+		summaryProvider:               summaryProvider,
+		nodeRef:                       nodeRef,
+		localStorageCapacityIsolation: true,
+		dedicatedImageFs:              &diskInfoProvider.dedicatedImageFs,
+	}
+
+	activePodsFunc := func() []*v1.Pod {
+		return pods
+	}
+
+	mgr.synchronize(diskInfoProvider, activePodsFunc)
```

Comment:
You should test the outcome of this function after `synchronize`.

Reply:
the outcome here is a list of pods, are you suggesting using ``reflect.DeepEqual()`` to compare if we evicted the right pod?

Reply:
Yea.  Make sure you test that it evicted the right pod.

Reply:
And Pod was evicted etc

Reply:
Please update this when you get a chance.  Maybe you can follow https://github.com/kubernetes/kubernetes/blob/a0873d37c58757ac6c2a339af1e4f8d601cdef20/pkg/kubelet/eviction/eviction_manager_test.go#L289 as an example for this.

Reply:
already updated. PTAL when you get a chance.